### PR TITLE
Add Vendor Detail v2 Api

### PR DESCRIPTION
**Pull request by poudelprakash**
- Add V2 Vendor Detail Api
- Add relevant tests

### CODE REVIEW

**Comment by poudelprakash** [comment]
@damianlegawiec could you please review this?

**Comment by damianlegawiec** [comment]
Thank you @poudelprakash could you also add request specs for the new endpoint?

eg. https://github.com/spree/spree/blob/master/api/spec/requests/spree/api/v2/storefront/products_spec.rb

File: app/serializers/spree/v2/storefront/vendor_serializer.rb [hunk 1]
```diff
@@ -4,7 +4,10 @@ module Storefront
       class VendorSerializer < BaseSerializer
         set_type :vendor
 
-        attributes :name, :about_us
+        attributes :name, :about_us, :slug, :state, :contact_us
```
Comment (damianlegawiec) [inline app/serializers/spree/v2/storefront/vendor_serializer.rb]: @poudelprakash let's not include state here

File: app/serializers/spree/v2/storefront/vendor_serializer.rb [hunk 2]
```diff
@@ -4,7 +4,10 @@ module Storefront
...
+
+        has_one :image
```
Comment (damianlegawiec) [inline app/serializers/spree/v2/storefront/vendor_serializer.rb]: @poudelprakash we need to declare a serializer for this

File: app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module SpreeMultiVendor
```
Comment (damianlegawiec) [inline app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb]: @poudelprakash we don't need this decorator anymore, already done

File: app/controllers/spree/api/v2/storefront/vendors_controller.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      module Storefront
+        class VendorsController < ::Spree::Api::V2::BaseController
+          def show
+            render_serialized_payload { serialize_resource(resource) }
+          end
+
+          private
+
+          def scope
+            Spree::Vendor
```
Comment (damianlegawiec) [inline app/controllers/spree/api/v2/storefront/vendors_controller.rb]:
```suggestion
            ::Spree::Vendor.active
```

File: app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb (hunk 3 above)
Comment (poudelprakash) [reply app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb]: I don't understand where it is already done..
Is it already merged?

File: app/serializers/spree/v2/storefront/vendor_serializer.rb (hunk 1 above)
Comment (poudelprakash) [reply app/serializers/spree/v2/storefront/vendor_serializer.rb]: @damianlegawiec Done, State has been removed

File: app/controllers/spree/api/v2/storefront/vendors_controller.rb (hunk 4 above)
Comment (poudelprakash) [reply app/controllers/spree/api/v2/storefront/vendors_controller.rb]: Done

File: app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb (hunk 3 above)
Comment (damianlegawiec) [reply app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb]: > I don't understand where it is already done..
> Is it already merged?

https://github.com/spree-contrib/spree_multi_vendor/blob/9619f3a68cb253c53d2ba7b697267e2d93c0ca47/app/serializers/spree_multi_vendor/vendor_models_decorator.rb#L1

File: app/serializers/spree/v2/storefront/vendor_serializer.rb (hunk 2 above)
Comment (poudelprakash) [reply app/serializers/spree/v2/storefront/vendor_serializer.rb]: @damianlegawiec it takes https://github.com/spree/spree/blob/master/api/app/serializers/spree/v2/storefront/image_serializer.rb
from the base,
any reason for this suggestion?

File: app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb (hunk 3 above)
Comment (poudelprakash) [reply app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb]: Awesome!

I have removed the decorator :)

File: app/serializers/spree/v2/storefront/vendor_serializer.rb (hunk 2 above)
Comment (damianlegawiec) [reply app/serializers/spree/v2/storefront/vendor_serializer.rb]: @poudelprakash it should work, but we use a different class here: https://github.com/spree-contrib/spree_multi_vendor/blob/master/app/models/spree/vendor_image.rb

File: spec/requests/spree/api/v2/storefront/vendors_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        expect(json_response.keys).to contain_exactly('data')
+      end
+    end
+
+    context 'with products and images included' do
+      before { get "/api/v2/storefront/vendors/#{vendor.slug}?include=products,image" }
+
+      it_behaves_like 'returns 200 HTTP status'
+
+      it 'returns product information' do
```
Comment (damianlegawiec) [inline spec/requests/spree/api/v2/storefront/vendors_spec.rb:38]: @poudelprakash let's also add a check for image attributes, thanks!

Comment (poudelprakash) [reply spec/requests/spree/api/v2/storefront/vendors_spec.rb:38]: I have now added a check for image attributes as well, [Please check here](https://github.com/spree-contrib/spree_multi_vendor/pull/156/files#diff-7bce2b451c4453d90192df7319a3e5f45ab01b9788f8ab578e24e9dac2847504R44) :)

File: app/serializers/spree/v2/storefront/vendor_serializer.rb (hunk 2 above)
Comment (poudelprakash) [reply app/serializers/spree/v2/storefront/vendor_serializer.rb]: I have now added serializer and spec for vendor_image, please check :)

File: spec/factories/spree/vendor_image_factory.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :vendor_image, class: Spree::VendorImage do
+    attachment { Rack::Test::UploadedFile.new("#{Rails.root}/../fixtures/thinking-cat.jpg", 'image/jpeg') }
```
Comment (poudelprakash) [inline spec/factories/spree/vendor_image_factory.rb]: @damianlegawiec I am new to image testing for engine,
Is there any better way to specify this path?
I had to use `Rails.root` to get dummy project path, and go one step outside with: `../` to reach fixtures folder.

Else the file would not be found.

Thank you :)

File: spec/serializers/spree/v2/storefront/vendor_image_serializer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    expect(subject.serializable_hash).to eq(
+      {
+        data: {
+          id: vendor_image.id.to_s,
+          type: :vendor_image,
+          :attributes => {
+            :viewable_id => vendor.id,
+            :viewable_type => vendor.class.name,
+            :styles => [
+              { :height => "48", :url => "/rails/active_storage/representations/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--8a45a5071ebace71ba86fe762e3b7a06ebcfc6c0/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCem9MWm05eWJXRjBTU0lJYW5CbkJqb0dSVlE2QzNKbGMybDZaVWtpQ3pRNGVEUTRQZ1k3QmxRPSIsImV4cCI6bnVsbCwicHVyIjoidmFyaWF0aW9uIn19--ab6fb539515f81f7d5e69715bc961d15d9d43f48/thinking-cat.jpg", :width => "48" },
```
Comment (poudelprakash) [inline spec/serializers/spree/v2/storefront/vendor_image_serializer_spec.rb:21]: also, I didn't see image related serializer specs in other spree projects,

please let me know if there is room for improvement in this spec as well :)

**Comment by poudelprakash** [comment]
@damianlegawiec Ready for review :)

File: app/serializers/spree/v2/storefront/vendor_image_serializer.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+module Spree
+  module V2
+    module Storefront
+      class VendorImageSerializer < BaseSerializer
+        set_type :vendor_image
+
+        attributes :viewable_type, :viewable_id, :styles
```
Comment (damianlegawiec) [inline app/serializers/spree/v2/storefront/vendor_image_serializer.rb]: @poudelprakash we can skip type/id

```suggestion
        attributes :styles
```

File: spec/factories/spree/vendor_image_factory.rb (hunk 6 above)
Comment (damianlegawiec) [reply spec/factories/spree/vendor_image_factory.rb]: https://github.com/spree/spree/blob/fd0131aea17c3a25972cf5079288bec7bd6870bc/backend/spec/controllers/spree/admin/stores_controller_spec.rb#L7 probably sth like this :)

**Comment by poudelprakash** [comment]
@damianlegawiec I have fixed the pr as per suggestions, Please review :)

**Review by damianlegawiec — APPROVED** [review]
Thank you @poudelprakash , awesome work!

**Pull request merged**